test/compilable/test14835.d

```diff
@@ -0,0 +1,15 @@
+// https://issues.dlang.org/show_bug.cgi?id=14835
+
+
+void reachIf(bool x)()
+{
+    if (!x)
+        return;
+    return;
+}
```

Comment:
```suggestion
bool reachIf(bool x)()
{
    static if (!x)
        return false;
    int a;
    assert(a == 0);
    return true;
}
```

The issue was explicitly about `static if`, but the test here only includes `if`.
Additionally, the extra statements are to guarantee that "Statement is not reachable" is triggered, which could stop happening to an empty `return` at the end of a `void` function due to unrelated changes.

Reply:
The solution to the `static if` issue is to change the behavior of `if`. Hence the test is about `if`.